src/algorithm/mod.rs

```diff
@@ -4,6 +4,10 @@ pub mod centroid;
 pub mod contains;
 /// Checks if the geometry A intersects the geometry B.
 pub mod intersects;
+/// Construct the intersections of geometries.
+pub mod intersection;
+/// Implements total ordering on points
+pub mod point_ord;
```

Comment:
where does `point_ord` get used in the `intersection` `impl`? i'm a little confused how both of these modules relate to each other

Reply:
It currently is used [here](https://github.com/georust/rust-geo/pull/122/files#diff-aff11658469bb534220beba341bdef9bR72) to make computing the intersection of two parallel, but potentially overlapping line segments easier. The basic idea is that if you sort the endpoints while keeping track of the line segment to which each endpoint belongs, you can figure out the intersection by looking at their order. For example if the sorted endpoints look like `[l1a, l2b, l1b, l2a]` then you know (provided none of them are equal) that there is a line segment overlap from l2b to l2a.

Reply:
It will also appear again in the more general intersection algorithm which requires the endpoints to be stored in a binary search tree.